google/auth/impersonated_credentials.py

```diff
@@ -229,3 +243,108 @@ def _update_token(self, request):
             principal=self._target_principal,
             headers=headers,
             body=body)
+
+    def sign_bytes(self, message):
+
+        iam_sign_endpoint = _IAM_SIGN_ENDPOINT.format(self._target_principal)
+
+        body = {
+            "payload": base64.b64encode(message),
+            "delegates": self._delegates
+        }
+
+        headers = {
+            'Content-Type': 'application/json',
+        }
+
+        authed_session = AuthorizedSession(self._source_credentials)
+
```

Comment:
You can mock out `AuthorizedSession.post` and `base64.b64decode` and assert that they've been called the correct # of times with the correct arguments. See similar examples in [test_id_token.py](https://github.com/googleapis/google-auth-library-python/blob/master/tests/oauth2/test_id_token.py).

Reply:
added in coverage and mocked the authorizedsession (i think; PTAL).

also, i dindt need to mock `b64decode`...while i could use the ' id_token.verify_token(idt,request)` within the testcode, its a bit strange since its not really verifying any testcase)